qiskit_nature/problems/second_quantization/electronic/electronic_structure_problem.py

```diff
@@ -67,15 +68,18 @@ def second_q_ops(self) -> List[SecondQuantizedOp]:
             total particle number operator, total angular momentum operator, total magnetization
             operator, and (if available) x, y, z dipole operators.
         """
-        self._molecule_data = cast(QMolecule, self.driver.run())
+        driver_result = self.driver.run()
         if self._legacy_transform:
-            qmol_transformed = self._transform(self._molecule_data)
+            qmol_transformed = self._transform(driver_result)
             self._properties_transformed = (
                 ElectronicStructureDriverResult.from_legacy_driver_result(qmol_transformed)
             )
+        elif isinstance(driver_result, QMolecule):
+            self._properties_transformed = (
+                ElectronicStructureDriverResult.from_legacy_driver_result(driver_result)
+            )
         else:
-            prop = ElectronicStructureDriverResult.from_legacy_driver_result(self._molecule_data)
-            self._properties_transformed = self._transform(prop)
+            self._properties_transformed = self._transform(driver_result)
```

Comment:
This still needs to be cleaned up. In order for that to happen I need to figure out how to properly deprecate the `BaseProblem.molecule_data*`..
@woodsp-ibm @manoelmarques do you have some ideas on how to proceed with that?

Since they are part of the public API I am unsure if I can just change/delete them. They won't exist in their format any longer and they were really just auxiliary access points to the driver results (which are now `Property` objects).
In #263 I added the `BaseProblem.roperties_transformed` attribute as a temporary workaround. Ideally I think we should have something like this:
```python
class BaseProblem:
    @property
    def driver_result(self) -> GroupedProperty:
        return self._driver_result

    @property
    def transformed_driver_result(self) -> GroupedProperty:
        return self._transformed_driver_result
```
Please let me know your opinions :slightly_smiling_face:

Reply:
You can’t delete the properties if you intend to deprecate. Is it possible to get the molecule data by other means to keep the properties or is it lost ?

Reply:
`molecule_data` used to be `QMolecule` or `WatsonHamiltonian`. Since these are now deprecated, they will no longer occur within the stack, if a user uses one of the new drivers in `qiskit_nature.drivers.second_quantization`.
I could deprecate the properties and change their type to be `Optional[...]` but I am not sure if that is allowed? [Technically they probably should have been that though, because the `__init__` sets them to `None`...]

And to be clear: the deprecated drivers are still supported by this stack but right now they will not store these properties (which I could change ofcourse) :+1:

Reply:
It would be nice if using things with the existing drivers, for users that have code to what is currently released, works the same albeit deprecated. Switching to the newer drivers then other code for properties can come about. At that level things can be more separated - in the problem I guess it has the unenviable task of in the interim allowing the old way to work and supporting the new way too if thats indeed doable.

Reply:
I have updated the code and we now have the following:
* `BaseProblem.molecule_data*` are deprecated in favor of `BaseProblem.grouped_property*`

We have some internal logic as follows:

#### Legacy Driver + Legacy Transformers
Works as expected. Deprecation warnings will be raised by those classes themselvers.
The following properties will be set:
* `molecule_data`
* `molecule_data_transformed`
* `grouped_property`
* `grouped_property_transformed`

#### Legacy Driver + New Transformers
Works. A `UserWarning` will be raised, suggesting to look into the new drivers.
The following properties will be set:
* `molecule_data`
* ~~`molecule_data_transformed`~~ (remains `None`)
* `grouped_property`
* `grouped_property_transformed`

#### New Driver + Legacy Transformers
Impossible! A `QiskitNatureError` will be raised.

#### New Driver + New Transfomrers
Works as expected.
The following properties will be set:
* ~~`molecule_data`~~ (remains `None`)
* ~~`molecule_data_transformed`~~ (remains `None`)
* `grouped_property`
* `grouped_property_transformed`


Finally, if a user mixes legacy and new transformers, a `QiskitNatureError` will
be raised, too.